Addon-docs: Prettier, collapsible values in ArgsTable

ajkl2533 opened the pull request:
Issue: #11681

## What I did
To make values in prop table more obvious and clean, I wrapped all values in a code-like wrapper. For long enums with more than 8 values, I introduced expanding wrapper, where the first 8 values are inlined with a "Show *n* more..." button. When the button is clicked all values are expanded and each value is on its own row.

<img width="860" alt="Screenshot 2020-08-03 at 10 09 38" src="https://user-images.githubusercontent.com/18654425/89164058-daaf4c00-d576-11ea-85ef-eb10e9d5e0bb.png">
<img width="860" alt="Screenshot 2020-08-03 at 10 10 00" src="https://user-images.githubusercontent.com/18654425/89164054-d97e1f00-d576-11ea-9e20-458a079f1637.png">


## How to test

- Is this testable with Jest or Chromatic screenshots? yes, Chromatic
- Does this need a new example in the kitchen sink apps? no
- Does this need an update to the documentation? no

ajkl2533 committed:
feat(ArgValue): pretiffy output of values in PropsTable

Ref #11681

shilman commented:
This looks so good @ajkl2533 !!! @domyen can you visually review?

domyen commented:
This is awesome! I accepted all the updates.

One last thing, can you create a story for the expanded state where each item has its own row? 

Another contributor could accidentally change the code and break this behavior so it's good to save it as a test baseline now.

domyen reviewed (changes requested):
This is awesome! I accepted all the updates.

Can you create a story for the expanded state where each item has its own row?Another contributor could accidentally change the code and break this behavior so it's good to save it as a test baseline now.

![image](https://user-images.githubusercontent.com/263385/89214785-70f26a80-d595-11ea-9064-da10ed73d469.png)

ajkl2533 committed:
feat(ArgsTable): add prop for default expansion of args

ajkl2533 commented:
@domyen Sure thing, I added new `hasExpandedArgs` prop to ArgsTable which can override the default behaviour and render all arguments and added story for this.

domyen commented:
lgtm! @shilman can you confirm? ready to merge from my end ✅

shilman reviewed (commented):
@ajkl2533 Sorry, slow to review the code. This looks great visually and I'm super excited about the improvement. However, I'm curious what's the rationale for the `hasExpandedArgs` flag? Can we just make it always be false to simplify the code?

ajkl2533 commented:
So one thing is I needed somehow show the expanded state in UI tests :D the second thing is that maybe someone would want to opt-out the expanding feature, which actually would be just for those who include the ArgsTable manually. What is your opinion @shilman I personally don't have a strong opinion on this and I'm ok to remove it, but then I'm not sure how to simulate expanded state in UI tests

shilman commented on lib/components/src/blocks/ArgsTable/ArgValue.tsx:
```diff
@@ -72,18 +99,52 @@ const calculateDetailWidth = memoize(1000)((detail: string): string => {
 });
 
-const ArgSummary: FC<ArgSummaryProps> = ({ value }) => {
+const uniq = (arr: string[]): string[] =>
```
can we use `lodash/uniq`?

ajkl2533 replied:
sure thing

ajkl2533 committed:
refactor: replace uniq with lodash

shilman reviewed (approved):
SWEET!! 💯
